lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -30,6 +31,92 @@ def post_sql_statements( table_name, options ) # :nodoc:
     end
   end
 
+  # Add a column to be updated on duplicate key update
+  def add_column_for_on_duplicate_key_update( column, options={} ) # :nodoc:
+    arg = options[:on_duplicate_key_update]
+    if arg.is_a?( Hash )
+      columns = arg.fetch(:columns) { arg[:columns] = [] }
+      case columns
+      when Array then columns << column.to_sym unless columns.include?( column.to_sym )
+      when Hash then columns[column.to_sym] = column.to_sym
+      end
+    elsif arg.is_a?( Array )
+      arg << column.to_sym unless arg.include?( column.to_sym )
+    end
+  end
+
+  # Returns a generated ON CONFLICT DO UPDATE statement given the passed
+  # in +args+.
+  def sql_for_on_duplicate_key_update( table_name, *args ) # :nodoc:
+    arg = args.first
+    if arg.is_a?( Array ) || arg.is_a?( String )
+      arg = { :columns => arg }
+    end
+    return unless arg.is_a?( Hash )
+
+    constraint_name = arg[ :constraint_name ]
+    conflict_target = arg[ :conflict_target ]
+    columns         = arg.fetch( :columns, [] )
+
+    sql = " ON CONFLICT "
+    conflict_target = sql_for_conflict_target( conflict_target, constraint_name )
+
+    if columns.respond_to?( :empty? ) && columns.empty?
```

Comment:
Yes, but the code currently looks to see if the table has updated_on or updated_at timestamp columns and if it does those are automatically added to the list of columns, so it's unlikely `columns` will empty. I'm thinking this code was put in place for MySQL since its on duplicate key update clause does not have a "do nothing" option, so columns were required. That brings up a good question, since Postgres does have a "DO NOTHING" option, should we not automatically add the update columns? @zdennis what are your thoughts?